primitives/src/vid/payload_prover.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Trait for additional functionality in Verifiable Information Retrieval (VID)
+//! to make and verify a proof of correctness of an arbitrary sub-slice of data
+//! from a payload.
+
+use super::{VidResult, VidScheme};
+use ark_std::ops::Range;
+
+/// Payload proof functionality for [`VidScheme`].
+pub trait PayloadProver<PROOF>: VidScheme {
+    /// Compute a proof for a sub-slice of payload data.
+    fn payload_proof<B>(&self, payload: B, range: Range<usize>) -> VidResult<PROOF>
+    where
+        B: AsRef<[u8]>;
+
+    /// Verify a proof made by `payload_proof`.
+    fn payload_verify<B>(
+        &self,
+        chunk: B,
+        commit: &Self::Commit,
```

Comment:
I think `range` should also be a public input here. The statement being verified is not just `chunk` is a slice from `commit`, but `chunk` is _the_ slice at position `range` in `commit`.

Reply:
It's assumed that the `range` data is included in `proof`. Both impls for this trait include a `chunk_range: Range<usize>,` field. It's sufficient to pass tests. I prefer it this way to reduce clutter in the args. Sound ok to you?

Reply:
I think morally it should be an argument because it's part of the statement being proven, conceptually, not part of the proof. Otherwise as a verifier, I need some way to extract the `chunk_range` from the proof to check it against the expected range, which needs to be incorporated into a trait somehow to make sense with the rest of the trait-based interface. And, we introduce a footgun where the verifier forgets to do this extra check, which is much harder to do if the expected range is a required parameter.

Reply:
New `Statement` struct in https://github.com/EspressoSystems/jellyfish/pull/389/commits/f399a4fbc7903b3b9d1c39df86ae7e248b6cd893

It's not needed in `payload_proof` but it does improve the readability of `payload_verify`.

I'll leave it to you to resolve this conversation when you're satisfied.